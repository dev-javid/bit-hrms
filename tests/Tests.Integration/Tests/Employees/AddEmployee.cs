using Application.Employees.Commands;

namespace Tests.Integration.Tests.Employees
{
    public class AddEmployee(TestWebApplicationFactory<Program> factory) : IntegrationTest(factory)
    {
        private const string Route = "/api/employees";

        [Fact]
        [UseReporter(typeof(DiffReporter))]
        public async Task Should_Return_Data()
        {
            await LoginAsCompanyAdminAsync();
            var command = new AddEmployeeCommand
            {
                DepartmentId = 3000,
                PersonalEmail = "abc@gmail.com",
                Aadhar = "123456789012",
                DateOfBirth = new DateOnly(2001, 06, 12),
                DateOfJoining = new DateOnly(2019, 06, 12),
                Address = "test",
                PAN = "GNCPS7000S",
                City = "test",
                PhoneNumber = "1234567890",
                JobTitleId = 4000,
                CompanyEmail = "abc@gmail.com",
                FatherName = "raza",
                FirstName = "mohd",
                LastName = "Saqib"
            };
            HttpResponseMessage response;
            using (StringContent content = new StringContent(JsonSerializer.Serialize(command), Encoding.UTF8, JsoneMediaType))
            {
                response = await Client.PostAsync(Route, content);
            }

            var result = await response.Content.ReadAsStringAsync();
            response.Dispose();
            Approvals.VerifyJson(result.Ignore("$.value.employeeId"));
        }
    }
}
