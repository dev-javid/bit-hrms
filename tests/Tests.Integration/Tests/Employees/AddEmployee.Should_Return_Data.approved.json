{
    "value": {
        "employeeId": "__IGNORED_VALUE__",
        "departmentId": 3000,
        "departmentName": "Test-Department",
        "jobTitleId": 4000,
        "jobTitleName": "Test-Department",
        "firstName": "mohd",
        "lastName": "Saqib",
        "fullName": "mohd Saqib",
        "dateOfBirth": "2001-06-12",
        "dateOfJoining": "2019-06-12",
        "fatherName": "raza",
        "phoneNumber": "1234567890",
        "companyEmail": "abc@gmail.com",
        "personalEmail": "abc@gmail.com",
        "address": "test",
        "city": "test",
        "pan": "GNCPS7000S",
        "aadhar": "123456789012",
        "documents": []
    },
    "formatters": [],
    "contentTypes": [],
    "declaredType": null,
    "statusCode": 200
}